### PR TITLE
New debian-sid-64 and debian-9-64 images for testing

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change adds 2 new images to the snapd tests execution.
Also, some tests have been modified to include/exclude properly the systems.

### CODE REVIEW

**Comment by zyga** [comment]
Can we please merge @mvo5's debian fixup PR here first: https://github.com/snapcore/snapd/pull/4256

File: tests/main/interfaces-many/task.yaml [hunk 1]
```diff
@@ -13,9 +13,10 @@ details: |
 # - Ubuntu classic 16.04 amd64 VM
 # - Ubuntu classic 18.04 amd64 VM
 # - All Ubuntu autopkgtests
-# - Debian unstable amd64 VM
+# - Debian sid amd64 VM
+# - Debian 9 amd64 VM
 # - TODO: All Fedora systems (for classic-only; unrelated error elsewhere)
-systems: [ubuntu-core-16-64, ubuntu-14.04-32, ubuntu-16.04-64, ubuntu-18.04-64, ubuntu-*-amd64, ubuntu-*-armhf, ubuntu-*-arm64, ubuntu-*-i386, ubuntu-*-ppc64el, debian-unstable-64]
+systems: [ubuntu-core-16-64, ubuntu-14.04-32, ubuntu-16.04-64, ubuntu-18.04-64, ubuntu-*-amd64, ubuntu-*-armhf, ubuntu-*-arm64, ubuntu-*-i386, ubuntu-*-ppc64el, debian-sid-64, debian-9-64]
```
Comment (bboozzoo) [inline tests/main/interfaces-many/task.yaml]: This can be converted to a list:
```
systems:
  - ubuntu-core-16-64
  - ubuntu-14.04-32
   ...
```

**Comment by zyga** [comment]
So this is with the new apparmor rules and with the new kernel and we are getting:

```
[Wed Nov 22 07:07:33 2017] audit: type=1400 audit(1511334453.993:15): apparmor="DENIED" operation="ptrace" profile="/usr/lib/snapd/snap-confine" pid=786 comm="snap-confine" requested_mask="trace" denied_mask="trace" peer="/usr/lib/snapd/snap-confine"
```
Perhaps there's a packaging problem somewhere. I'll look into this on my Debian installation.

File: spread.yaml [hunk 2]
```diff
@@ -66,7 +66,10 @@ backends:
                 kernel: Direct Disk
                 image: ubuntu-16.04-64
                 workers: 4
-            - debian-unstable-64:
+            - debian-sid-64:
```
Comment (mvo5) [inline spread.yaml]: We should also include this in the `qemu` section.

File: tests/main/interfaces-many/task.yaml (hunk 1 above)
Comment (mvo5) [reply tests/main/interfaces-many/task.yaml]: Wwe could use `debian-*` here instead of listing both explicitly.

**Comment by chipaca** [comment]
Is the dropping of all those `quiet` wrappers intentional?

**Comment by mvo5** [comment]
The upgrade test not working is kind of expected, the old version of snapd will not work on debian-sid.

**Comment by sergiocazzolato** [comment]
@chipaca yes, it is intentional, I am restoring this in next push.

File: tests/lib/mkpinentry.sh [hunk 3]
```diff
@@ -11,6 +11,9 @@ else
             ;;
         *)
             mkdir -p ~/.snap/gnupg/
+            # It is just created once by gpg-agent, then if this dir does not exist gpg will
+            # fail to create the key
```
Comment (zyga) [inline tests/lib/mkpinentry.sh]: Woah, is there a bug for this?

Comment (sergiocazzolato) [reply tests/lib/mkpinentry.sh]: @zyga gpg v2.1 says in its doc that that dir is created just once, then at some point it was changed and in the version 2.2.20 (the one used in debian-sid) if the dir does not exist, it is created automatically by the gpg-agent.

**Pull request merged**